### PR TITLE
Implement Bulk TSSC Transfer in DTP CLI via Fund contract

### PR DESCRIPTION
**Pull request by abhi3700**
## Description
Implemented a new `Fund` contract to facilitate bulk transfers of TSSC tokens with a single signature from the funder, improving upon the traditional method of signing each transfer individually using the `eth_sendRawTransaction` chain API call in `transfer_tssc_single`.

This update enables all transfers to be added in a single transaction within a block, as opposed to the previous iterative approach used in the `transfer_tssc_single` function, which resulted in multiple transactions being added to different blocks with a gap of 1-5 blocks between each transfer.

## Key changes
- [Introduced a Fund contract](https://github.com/abhi3700/subspace-evm-playground/compare/d1da2720fa4576103414e830a56b356b746625b7...d6346ddc199cfc66f803bbd76ac9e54152fb86db#diff-3e6016c675e31e69d5a1a6a624735fd8142a34282dbb47b9b39422aa619a79cb) for bulk TSSC token transfers with a single funder signature, replacing the previous iterative `eth_sendRawTransaction` approach.
- Debugged [`transferTsscToMany`](https://github.com/abhi3700/subspace-evm-playground/compare/d1da2720fa4576103414e830a56b356b746625b7...d6346ddc199cfc66f803bbd76ac9e54152fb86db#diff-3e6016c675e31e69d5a1a6a624735fd8142a34282dbb47b9b39422aa619a79cbR36-R48) for multiple receivers, added corresponding contract [tests](https://github.com/abhi3700/subspace-evm-playground/compare/d1da2720fa4576103414e830a56b356b746625b7...d6346ddc199cfc66f803bbd76ac9e54152fb86db#diff-63357471d2fb30b4517087c3ec467a3587bb761eacb505c1962b627788a90343), [deploy script](https://github.com/abhi3700/subspace-evm-playground/compare/d1da2720fa4576103414e830a56b356b746625b7...d6346ddc199cfc66f803bbd76ac9e54152fb86db#diff-dd3e0aa5776d459b525ce552f3aa69b088e3113c6925f8b9efa5fec5921f2786) and modified the function accordingly.
- Kept [`transferTsscToSingle`](https://github.com/abhi3700/subspace-evm-playground) as a private method and called within [`transferTsscToMany`](https://github.com/abhi3700/subspace-evm-playground/compare/d1da2720fa4576103414e830a56b356b746625b7...d6346ddc199cfc66f803bbd76ac9e54152fb86db#diff-3e6016c675e31e69d5a1a6a624735fd8142a34282dbb47b9b39422aa619a79cbR36-R48).
- [Renamed `transfer_tssc` to `transfer_tssc_single`](https://github.com/abhi3700/domain-transaction-producer/pull/2/files#diff-c730d471704d7db227e676d88503447de9b5463c8206b949b66dbb6da2211073R373) in **dtp** package of the workspace.
- Added `transfer_tssc_bulk` function in **dtp** package for single-transaction bulk transfers, and archived the [now-obsolete `transfer_tssc_single` function](https://github.com/abhi3700/domain-transaction-producer/pull/2/files#diff-c730d471704d7db227e676d88503447de9b5463c8206b949b66dbb6da2211073R342-R346).
- [Added `FUND` contract address into `.env.example` file](https://github.com/abhi3700/domain-transaction-producer/pull/2/commits/5723c8319aed906239474b2dcdca8645d6e01066).

### CODE REVIEW

File: dtp/src/utils.rs [hunk 1]
```diff
@@ -404,15 +339,153 @@ pub(crate) async fn gen_wallets_transfer_tssc(
...
+            funding_amount
+                .checked_mul(U256::from(tos.clone().len()))
+                .expect("Error in multiplying fund amount w receivers len."),
+        )
+        .send()
+        .await
+        // FIXME: Failure in getting pending tx: Revert(Bytes(0x))
+        .expect("Failure in getting pending tx")
+        .await?
+        .expect("Failure in \'transferTsscToMany\' of Fund contract");
```
Comment (0xjeremyfrank) [inline dtp/src/utils.rs]:
```suggestion
        .expect("Failure in \'transferTsscTooMany\' of Fund contract");
```

File: dtp/src/utils.rs [hunk 2]
```diff
@@ -404,15 +339,153 @@ pub(crate) async fn gen_wallets_transfer_tssc(
...
+
     Ok(signers)
 }
 
+/// Transfer TSSC to single account via eth API call method
+/// NOTE: For multiple receiver accounts, we can also use
+/// the contract "Fund"'s `transferTsscToMany`
+/// One can also leverage the contract "Fund"'s `transferTsscToSingle`
+/// function.
+pub(crate) async fn transfer_tssc_single(
```
Comment (0xjeremyfrank) [inline dtp/src/utils.rs]: this can probably be removed.

**Review by 0xjeremyfrank — COMMENTED** [review]
The bulk transfer works great! I was able to fund > 1,000 accounts before hitting gas limit. I had a couple of comments that should be addressed.

However, sending subsequent transactions doesn't work at scale, it breaks when passed 1000 accounts with a "Too many connections" error. I'm not sure the underlying cause, but much of the code seems inefficient in the # of calls made. These inefficiencies are outside of the code in this PR but include calling  `get_chainid` in every iteration, and in `get_gas_cost` multiple calls to `get_transaction`. I didn't do an exhaustive look, but I think for the next PR you should try to make the code much more efficient and test at a bit higher scale locally.

File: dtp/src/utils.rs (hunk 1 above)
Comment (abhi3700) [reply dtp/src/utils.rs]: Here, are you suggesting to change the function name of the contract or just as error message?
Actually, I referred to `transferTsscToMany` as the contract method being called here.

File: dtp/src/utils.rs (hunk 2 above)
Comment (0xjeremyfrank) [reply dtp/src/utils.rs]: I mean remove the function completely, I don't think we will use it.

**Comment by abhi3700** [comment]
Noted!

For the error: "Too many connections", I can use batching with chunk size of 100 requests. I tested the current version of the code locally with different account numbers - `100`, `200`, `300`, `400` async calls. This is my observation:

- `100`: 100% success. Like here, 100 txns got added in 2 blocks.
  <details><summary><b>Terminal output:</b></summary>

  ```sh
  =====================
  Sending Light txs...
  =====================
  
  Number stored in 'Counter' before calls: 4069
  
  '0xde16…a6ad' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x52cc7d287043a240579ee0ebc46f6208db60b12f2b639b26be2bfe1e6c0a6c2e', indexed at #22 in block #57566.
  
  '0x8666…317e' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xece15f1d767e8449632e6cfda77cd8222dcb9bc634cb1e64251c63a0451c66f7', indexed at #57 in block #57566.
  
  '0xc878…6592' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x0df2e3218ddd8add7e379aea22e7865a5c97285983793639a7045f765cc767e9', indexed at #24 in block #57566.
  
  '0x6a98…93bd' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xc7051f465a1bd5b68f9119d7cd2c0888a86e4343186e2a0ff0768b0d01003a7f', indexed at #48 in block #57566.
  
  '0x9f82…b948' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x9bd9cc641cfbeed81b9cf7b690b56f0cafc181957dae60fd8f563dc7d6df68f6', indexed at #41 in block #57566.
  
  '0x8fe4…5e7a' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x1d47dab110f93150eb4497abf2ef5a5eb17079cdba48dc2d15ee426a83ed75a8', indexed at #70 in block #57566.
  
  '0x79b6…ee30' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x3be1af071302a0d759f5573169f61cf5a0402c3f793833c973389e6829794c96', indexed at #72 in block #57566.
  
  '0x51d4…06a0' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xd8c7fa485c3a1893c10faa19bc4d5fafb5b4b09affe0dac5d29205bbe7ea7a56', indexed at #0 in block #57566.
  
  '0x746f…9bc7' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x4880b66718831e904ea6bef2d06bbc899cf50e412d0461b23ba707754c62ea84', indexed at #62 in block #57566.
  
  '0x33c4…13dd' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x7755bb5b7cbcd345815f0706aadcab238a4f8a0cc2479ff780eb9c6ee28ae219', indexed at #60 in block #57566.
  
  '0xd1e7…b245' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x745fb87a73604c59cda53b2966d9de0abe1cbdecfc5412ba0fdf4654f4d3ccf8', indexed at #68 in block #57566.
  
  '0x8961…3c53' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xa5a2bb06cfa12055b0bd558d4047320fdf6e504442c87c4a69244acd06937bd8', indexed at #47 in block #57566.
  
  '0x0acd…4020' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xa6398875a35e6e781b7b74f349f3c967f4f4ef702e2ac52bbfda4cfb4a07c552', indexed at #61 in block #57566.
  
  '0x7dce…e2db' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x9b3e78191a4579374a80ab436785d09c3b8ce0af589d32cd571abb1e2a1c7087', indexed at #9 in block #57566.
  
  '0xa230…fd42' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xea294d1818527fc14492ad5998e1ffd107221ba01071eeb0c77ddd82171044aa', indexed at #17 in block #57566.
  
  '0x8ee2…1fb8' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xd619088f55288264cbdf7c12fb0532fa152a9657833d558eec0f99304f89544c', indexed at #39 in block #57566.
  
  '0x7e05…2c10' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x893bde47008b35179e8f402697ccae7679846105922835988400cb8d8aa44a87', indexed at #3 in block #57566.
  
  '0x7167…2ecb' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x1881d9cb7951bfd84fdc0ad4ef55bf70e412ba4371533ec94c2ed149ad8dd25f', indexed at #2 in block #57566.
  
  '0x0afe…d1aa' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x99480837ca94723b6077dd1bfa24e9ba7f0322fa967b67e653d541883da4bada', indexed at #56 in block #57566.
  
  '0x869f…c718' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x5b3160c0cb29de118b7b4c6eeb24c6ae13067fa16708e459fd0bed6ce4e16732', indexed at #10 in block #57566.
  
  '0x21c8…72b6' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x5dce4ff2415022d3894520d9f4bfa5eaaca8a5ebe06fc9cd9985f522808f0d3b', indexed at #12 in block #57566.
  
  '0x179c…feb5' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x317cf65488bd4963e66e5ec9e38d1ac3e49f416e72d7248ae9839fe0695edd0c', indexed at #28 in block #57566.
  
  '0xf880…1f89' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x2e33d4de8cc9be9773feb70b7e5e5c3ff10b584d6dc6968077b71de3fa8e8661', indexed at #51 in block #57566.
  
  '0xc8f1…ef9a' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x6a177bf1e573b174ddb3a194f5c863be6e2db2217816aa87dcccc24d1d2cfddf', indexed at #37 in block #57566.
  
  '0x4212…0c7b' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x8c621f0628d6ba1f18eecaedffdf072f63b0aa1e244a6c25bc07fbcee65cf8a0', indexed at #67 in block #57566.
  
  '0x882e…defd' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xdc59043a837289e6ee4e450c7efd04bd0c0c863d6d49c684deff1b6df7cbeb51', indexed at #23 in block #57566.
  
  '0x47de…7933' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x7278351e2f49b8164b999b66f976bcc4f11a001d1da33be3bb89d7cea0214c45', indexed at #30 in block #57566.
  
  '0xe463…edd2' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xb502519b37bedc987d2b4ef928b8bfbe45545913d558fe9b5e02896a6c849635', indexed at #43 in block #57566.
  
  '0xdf46…ce32' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xea19874f45132dff7b43e4dab796c45264627f82237aaea7471161a77d8bf92a', indexed at #4 in block #57566.
  
  '0x410a…7426' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x419996f283236f27f926d849d72ed27c017021d0c6f2f575c55b16a0aa5a0b7a', indexed at #6 in block #57566.
  
  '0xeabd…d92c' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x4d5d3191a5fc6638743a44d434112dfa5cf76dddb19dedf9c0f87692fcf83e82', indexed at #55 in block #57566.
  
  '0x7a04…7e1d' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xff0ba55365e3af79b8851b4175ba4dd1fc22fb75c4725da077cd64fa696ec93e', indexed at #54 in block #57566.
  
  '0xdd21…0b64' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x70760e5db73f4b31e2405a4ca3956a5e50c00fca5721596a71fd549a3863104a', indexed at #8 in block #57566.
  
  '0xa9f0…42c7' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x4c5918ed9c0b10ccadf38a8742c610484e5d9c261ff00fd02940fc2beea9913c', indexed at #16 in block #57566.
  
  '0xcdbb…90a0' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xd9accfef5b6d50fa9942c6ede625d756c64b62dd408579a479b3037a9853854b', indexed at #35 in block #57566.
  
  '0x705e…bfe4' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xb4da4977d1eb94ff8ab1325b096e2f971fcecccc4884b9e5f600202abb177f5c', indexed at #49 in block #57566.
  
  '0x1f8c…fbd1' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x03870e99d6e445c24d543a886ccda5e5be8b3dcb1ddcb335b29044dd7acea247', indexed at #14 in block #57566.
  
  '0xcc31…59c1' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xaaa0eff083b59107406d6981655badf342fa335a7761c3cf746cdf84d20a51d5', indexed at #29 in block #57566.
  
  '0x0151…eadc' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xe7857569868462c358051897b6003118c432967eb3298e5b70ad5862b1c01d66', indexed at #64 in block #57566.
  
  '0xa4d6…ee0b' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xa14c6dd487cfab584670edba2d8dc0af85013a208f471ce3d7d7b377fe6a0e9f', indexed at #44 in block #57566.
  
  '0x31b3…c736' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xc68cb5bf7b4a0838ae6a9101bddcae16edc2835de6ce45277ae59d7d3d52ea63', indexed at #58 in block #57566.
  
  '0x8974…2d85' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x5080cd8d48e698f65d47d1a7dd4bb69b9bf3ec69756b2d3e0e7e83ecd116ea77', indexed at #5 in block #57566.
  
  '0xfd2b…6e09' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xe37be20219b72ea7be0282857ee40edbbc542cf6e4a578750cb9882179623e87', indexed at #45 in block #57566.
  
  '0x1b8f…644c' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xa68ab85cb8d85bf58a729fdecf3eca689a2e3b0cd93403d4cfc49a8e33d7ce35', indexed at #20 in block #57566.
  
  '0xba45…b908' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xe7a7b674de9c936d5fc4805e671d14d0029e44bbc1e2319a4dda8be3aa1b9e09', indexed at #19 in block #57566.
  
  '0xdcd2…2813' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x42a5bbe8f1f75e87b335db70f17d86d32d1f328fbb347a84d77432ab1475ff11', indexed at #13 in block #57566.
  
  '0xf366…00af' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x4b783d35a581c4603a1cd6d88eaac84c59bacc21efb83be40181a3dad0fb6275', indexed at #59 in block #57566.
  
  '0xfb1f…a047' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xfa2b2371c0c54337970c2f01bc6a371149ad1d2caedf939eaf708f4541d5b392', indexed at #27 in block #57566.
  
  '0x63f1…7c57' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x58bd062221c07cd065885cef7fb228776cfe8e34b64faaf98b7328c51b7cec6b', indexed at #26 in block #57566.
  
  '0x6d65…0cda' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xe4c2248508bdf7eb4cf7ae71542bbc3296581d0e0c7198eb164d03639fae2da6', indexed at #31 in block #57566.
  
  '0x24eb…1929' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xc4195bafc5d43fd848a6a0465b730c43d63ae564d7e804573246180548c77ca3', indexed at #46 in block #57566.
  
  '0x7521…e8c8' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xaae74368c200880d6dc193eebf21f0cc93e85af4663ea3816cc626b632a1ccb1', indexed at #25 in block #57566.
  
  '0xfdac…260e' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xe2d05835954129527e8c8ef6f0e56a06a88894ff89df8de32398c7b4ad4838f7', indexed at #42 in block #57566.
  
  '0x5b1d…603b' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xb2dff4e6d2dede3fe72858dc678a733cb9ea57d85e626e47ed42c9364e743de5', indexed at #15 in block #57566.
  
  '0x28a4…8455' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x9f97cbe46f1792b16699883b1f466fcaccefbfa025b1b3667ee66d0306529173', indexed at #36 in block #57566.
  
  '0x1621…b18a' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xcb98e2a8951cd6d38bf36b8ab950bbed590f2ccdfbc8fd6275225218bd5a2b13', indexed at #21 in block #57566.
  
  '0x4ab3…cf3d' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xe17bbf6b71a4bcb9f415e4a4174555b6ced8f308550a94f3864922708d1c74c7', indexed at #66 in block #57566.
  
  '0x1b8a…c855' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x7bd54e68afe879b79d846a5b29acb8b3c9405c7ccfb0c6eee13d4221540f79a4', indexed at #63 in block #57566.
  
  '0x3099…0af2' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xe9f792624c547c1f5e685c2afb2a79fdfeb8932d0625743fd5850120cf31e4d1', indexed at #32 in block #57566.
  
  '0xa0d4…3043' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xc03350de77db4107ecef2b4440a070e81d521ca23c3c573d346d1cff63412778', indexed at #53 in block #57566.
  
  '0xf9ef…0bc5' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x1642ee8dd70f11a2320ac475c76f2bac224a871ac04323dd4ea6dcadf87292a3', indexed at #7 in block #57566.
  
  '0x2d69…1902' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xb91f59f69f5ef714b831c7e0d403ef879bae3f36dd9469530331814724f5ed60', indexed at #50 in block #57566.
  
  '0x2fc8…a02e' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xdf8f699f36b980a4f94ea9745891ea455f4cc5c469e3083b611fc9c0261b75de', indexed at #38 in block #57566.
  
  '0x127f…7574' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x8645cbcf84f7cc6926df3cf15f7e7c5eb11dfce2540560bd46d62528edd64350', indexed at #18 in block #57566.
  
  '0x5f5f…d39b' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x32131131f32099f7e8f8c3d842e982ea70e21f3672601b08fa5af9c2e69315e9', indexed at #1 in block #57566.
  
  '0x52f9…6c85' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x3e89e53a642cb02e04196cf7aaaf62d075600b75496b73472cfd80787eefcaeb', indexed at #65 in block #57566.
  
  '0xd4e9…17db' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x2c6c804eeff8173e836075ea81720e3e958579370f5e52406af95bbfd0119a5a', indexed at #11 in block #57566.
  
  '0x0e34…6dfd' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xd651a5a1dba5210139b8a2987996cc1cd60fe34c18a77343a0c227195299772e', indexed at #40 in block #57566.
  
  '0xb500…ba96' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xf424a7d8d36e379b7accb5d8f0115384f77650d5397bacd8f0a2e2085a9c1f8c', indexed at #34 in block #57566.
  
  '0x57a4…ec91' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x173687061ea1dbbf87deb1f845cbb0b54deec5c38a8f6a15ae927b04c8e354ec', indexed at #52 in block #57566.
  
  '0xe8f0…c00f' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x7acee9be3e1cf4b3d293573bc315e2547a8eeddb391a581d652761c98bfcd14f', indexed at #71 in block #57566.
  
  '0x8798…a1ba' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x82d10dd85998b97c1d5a896610edf0aea485639384c51300a74686c107f7a958', indexed at #33 in block #57566.
  
  '0x68f6…f426' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x780ce5ea58a9549e8d9fdc6215d17c432b14d6f4dc29808d7dfc6977214dd8ac', indexed at #69 in block #57566.
  
  '0xb508…03a8' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x2217f638f36e7640f1a23fbbf1bc5ced927ad68a308a7299358aa2a69796a532', indexed at #21 in block #57567.
  
  '0xd923…aead' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x1b8c88a28dbce9f9c0aa54e5a4b5ea396648201ec78ecc516bd2f370aeffe8a4', indexed at #17 in block #57567.
  
  '0x9313…64a7' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x9721456b87345f42d85f5c64ebe2ab434a04cac129372f7831443f41af564295', indexed at #20 in block #57567.
  
  '0xae73…7068' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x22f7c08fdc81eb1409eb68ed82a48bcb0c1c850ff7857837ef6be8de7cfb88d6', indexed at #0 in block #57567.
  
  '0x58d8…282d' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x38f1cfaf07145b344cfb60fd6c918db924a9ea6d3194ac3dc0dcbd5527aa2f1a', indexed at #9 in block #57567.
  
  '0xf3a9…721c' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xbec43af8d0c445142e52728b741f2b2eb63810553be33da2a69e1b9266e5ebf9', indexed at #10 in block #57567.
  
  '0x5be7…4b9a' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x2a95fc8b901ccd6a38d5ca959a12063ec63edd224bac77240cd193e748df98e0', indexed at #5 in block #57567.
  
  '0x5f7e…bbb4' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xcc8e89b36cac812034a6e61376b9ae58b3fd6dd8505c70e24d491b0cd9342273', indexed at #3 in block #57567.
  
  '0x7063…eac7' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xd4fbcb88f2393b356c3c887cecc3c29f78b14c62fbc540e509752cc1f8915aab', indexed at #15 in block #57567.
  
  '0x5d99…b04e' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x0acb6df1306656d14a7480bd69c58d7d524bb0cb427b3aa12675af9af58c1838', indexed at #12 in block #57567.
  
  '0xff09…3e3e' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x91a625a7d6b78fa1fcd7cf7973a1269f252bca6936835f7a4a020b1fb36ad984', indexed at #14 in block #57567.
  
  '0xba54…6c31' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xad27b27441502f6db9cd7a2478b5dfb1529b0b5856049129db39fd16ef07bf94', indexed at #22 in block #57567.
  
  '0xe546…0804' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x21753b914caecaccba731b44e59bb1d0b86dc47f5aab53f5971a2f210fd2fde0', indexed at #13 in block #57567.
  
  '0x3424…ce9d' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x9340ad4dd75d35936be75c139fdc305606326167f121b02117eebd6420fb6a6a', indexed at #18 in block #57567.
  
  '0x898a…76a7' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x91195ccc1cc26c2fc30e3e8564daf7c4800ff7b72ea0d314be186744e49f776b', indexed at #25 in block #57567.
  
  '0x0d2a…97b0' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x7526e7aef876c43dbc7bb0b8d4314b376b7d5fe1dfe8afdda771023a1dab290c', indexed at #7 in block #57567.
  
  '0x4fe0…7ee6' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x4551d40cccc58732d00f678ee8b9397d5e8ce15e34a167fa3fff5f5a15b3ddc8', indexed at #24 in block #57567.
  
  '0xaa97…77b3' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xeabbe348b5822c1153442f49751e40dc49a9a20c0227444fb6e087ad1bcf85a0', indexed at #23 in block #57567.
  
  '0x21e2…17f8' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x2a300bf4e65cc0b11be463a58a3bcf5fce47b86b44034167a2781e4bcc82e422', indexed at #8 in block #57567.
  
  '0x269b…5310' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xb7f5af314634021e9cb03e713f1ffcec06353eb4181c2ac3797076e500e69221', indexed at #16 in block #57567.
  
  '0x8fc7…8a90' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x0df8e35f253533a5f783c4e9e5262204298e27d08e2e9ed9a43b1a07b6761625', indexed at #4 in block #57567.
  
  '0xaeb0…1def' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xe579cfdb3bbb0a6cf9926bf65d9685a124c1aa40c821f5f48143914912f5d336', indexed at #1 in block #57567.
  
  '0xf7dd…f387' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x59116f1165c1ca37b6ead700765724bd0e27c4455a25c47b44b5ead45929f577', indexed at #19 in block #57567.
  
  '0xb0b8…6f1b' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x603ab2f5f3eeeaa5f3ee1d6f9d2ffe15fafad0362ca03470dc0699bc3ebfb6c8', indexed at #6 in block #57567.
  
  '0xdd6a…5ccb' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x1873259b05db7310320c961b9e4251b0be4e1c8a5bf895dda0904a719dec23a5', indexed at #2 in block #57567.
  
  '0x7d73…a53a' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0x9117a34ba5cfc75e03718e15d2c254d628203cb73815a59e8d93e1a67a06b0a2', indexed at #11 in block #57567.
  
  '0x5148…417e' increment number, which incurred a gas fee of '0.00007928100052854 TSSC', has a tx hash: '0xfa3d427b9d84090b21eb920d2e3660c0e9f2bfedb8fb390bea04dacb1b188b81', indexed at #0 in block #57569.
  
  Number stored in 'Counter' after 100 calls: 4169
  ```
  </details>
- `200`: 70% success, not sure as sometimes it fails.
- `300`:  50% success, not sure as sometimes it fails.
- `400`: 100% failure

> Here, whenever failed, it gives the same error - "Too many connections".

So, I am thinking of considering `100` as max. chunk size to be run in loops. E.g. for `1000` accounts, possibly the light txs would be added in the next 10-20 blocks.

I will take this up as code changes in the next PR #3. 

For now, I will follow your code changes & some cleaning in this PR.

File: dtp/src/utils.rs (hunk 2 above)
Comment (abhi3700) [reply dtp/src/utils.rs]: Done in this [commit](https://github.com/abhi3700/domain-transaction-producer/pull/2/commits/5e4843e117e797fac9a346966289fdcdab91f9af#).

**Comment by abhi3700** [comment]
Addressed to the concerns in PR #3.
Ready to close this PR.

File: dtp/src/utils.rs (hunk 1 above)
Comment (0xjeremyfrank) [reply dtp/src/utils.rs]: nevermind, I think i misread this. ignore. 🤦‍♂️

**Review by 0xjeremyfrank — APPROVED** [review]

**Pull request merged**